src/SignalR/common/Protocols.MessagePack/src/Microsoft.AspNetCore.SignalR.Protocols.MessagePack.csproj

```diff
@@ -17,7 +17,7 @@
   <ItemGroup>
     <Reference Include="Microsoft.AspNetCore.SignalR.Common" />
     <Reference Include="MessagePack" />
-    <Reference Include="MessagePackAnalyzer" />
+    <Reference Include="MessagePackAnalyzer" PrivateAssets="All" />
```

Comment:
Why is this a `Reference` instead of a `PackageReference`? How does this even work? _does_ it even work? Have you tested introducing use of a banned API to see that an error is produced?

Reply:
![image](https://user-images.githubusercontent.com/2266487/77074773-9a996380-69f1-11ea-97ce-766ab75addc7.png)

Reply:
@tebeco can you dig out the `nuspec` generated for this package (the `nupkg` should be in the `artifacts` directory after a `.\build.cmd -pack`) to confirm that the `PrivateAssets` flag got propagated properly and there's no dependency on the analyzers in the final output? As you noted, we do some funky stuff with `Reference` nodes (I don't have the context as to why) so we should triple-check it's flowing properly to ensure we don't expose this package transitively.

Reply:
updated https://github.com/dotnet/aspnetcore/pull/19989#issue-390921135